spec/src/main/asciidoc/architecture.asciidoc

```diff
@@ -193,47 +193,442 @@ For example, calling an `@Outgoing` annotated method from user code will not pos
 
+This section lists the methods signatures that must be supported by the Reactive Messaging implementation.
 Implementations must validate that the stream shape matches the `@Outgoing` and `@Incoming` annotations, if they don't, a CDI definition exception should be raised to the CDI container during initialization.
 
-===== Simple method streams
+It's important to remember that users must not call these methods directly. 
+They are invoked by the Reactive Messaging implementation following the Reactive Streams protocol.
+
+Also the method must be implemented in a non-blocking fashion.
+For blocking transformations, asynchronous variants can be used.
+
+INFO: _assembly_ time is when the Reactive Messaging implementation initializes itself and creates the different bean instances and connects them together.
+
+INFO: In the following lists, `Message` can be an implementation of the `Message` interface.
+
+===== Methods producing data
+
+[cols="2a,1,1",options="header"]
+|===
+|Signature
+|Behavior
+|Invocation
+
+|
+[source, java]
+----
+@Outgoing("name") 
+Publisher<Message<O>> method()
+----
+| Returns a stream of `Message` associated with the channel `name`. 
+| Method called once at assembly time.
+
+|
+[source, java]
+----
+@Outgoing("channel") 
+Publisher<O> method()
+----
+| Returns a stream of _payload_ of type `O` associated with the channel `channel`. Produced payloads are mapped to `Message<O>` by the Reactive Messaging implementation.
+| Method called once at assembly time.
+
+|
+[source, java]
+----
+@Outgoing("channel") 
+PublisherBuilder<Message<O>> method()
+----
+| Returns a stream of `Message` associated with the channel `channel`. 
+| Method called once at assembly time.
+
+|
+[source, java]
+----
+@Outgoing("channel") 
+PublisherBuilder<O> method()
+----
+| Returns a stream of _payload_ associated with the channel `channel`. Produced payloads are mapped to `Message<O>` by the Reactive Messaging implementation.
+| Method called once at subscription time.
+
+|
+[source, java]
+----
+@Outgoing("channel") 
+Message<O> method()
+----
+| Produces a infinite streams of `Message` associated with the channel `channel`. 
+|	This method is called for each _request_ made by the subscriber.
+
+|
+[source, java]
+----
+@Outgoing("channel")
+O method()
+----
+| Produces an infinite streams of _payload_ associated with the channel `channel`. Produced payloads are mapped to `Message<O>` by the Reactive Messaging implementation.
+|	This method is called for each _request_ made by the subscriber.
+
+|
+[source, java]
+----
+@Outgoing("channel")
+CompletionStage<Message<O>> method()
+----
+| Produces a infinite streams of `Message` associated with the channel `channel`. The result is a `CompletionStage`. The method cannot be called until the `CompletionStage` returned previously is completed.
```

Comment:
Is this specifying "the method should not be called by a reactive messaging implementation until..."?